### PR TITLE
Update lingo #3

### PR DESCRIPTION
**Pull request by berniev**
Another round of fixing simple warnings and other minor improvements.
---

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6852">pipeline status</a> for feature branch [PR_6852](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6852). Pipeline [534035724](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/534035724) was triggered at [a1858b1](https://github.com/FreeCAD/FreeCAD/pull/6852/commits/a1858b1435b59c375732821a40494e666a8a8a35). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/App/Enumeration.cpp [hunk 1]
```diff
@@ -149,11 +149,11 @@ void Enumeration::setEnums(const std::vector<std::string> &values)
 
     _EnumArray = new const char*[values.size() + 1];
     int i = 0;
-    for (std::vector<std::string>::const_iterator it = values.begin(); it != values.end(); ++it) {
+    for (const auto & value : values) {
 #if defined (_MSC_VER)
         _EnumArray[i++] = _strdup(it->c_str());
```
Comment (chennes) [inline src/App/Enumeration.cpp:154]: You need to update the other half of this #if

File: src/App/Expression.cpp [hunk 2]
```diff
@@ -1580,11 +1581,9 @@ void OperatorExpression::_toString(std::ostream &s, bool persistent,int) const
     if (right->priority() < priority()) // Check on operator priority first
         needsParens = true;
     else if (rightOperator == op) { // Equal priority?
-        if (!isRightAssociative())
-            needsParens = true;
-        else if (!isCommutative())
+        if (!isRightAssociative() || !isCommutative())
             needsParens = true;
-    }
+            }
```
Comment (chennes) [inline src/App/Expression.cpp:1586]: You probably didn't mean to change this indentation.

File: src/App/Expression.cpp [hunk 3]
```diff
@@ -1797,7 +1796,7 @@ FunctionExpression::~FunctionExpression()
 
 bool FunctionExpression::isTouched() const
 {
-    std::vector<Expression*>::const_iterator i = args.begin();
+    auto i = args.begin();
```
Comment (chennes) [inline src/App/Expression.cpp:1799]: Lost the `const`-ness.

File: src/App/Expression.cpp [hunk 4]
```diff
@@ -2463,7 +2462,7 @@ Expression *FunctionExpression::_copy() const
 
 void FunctionExpression::_visit(ExpressionVisitor &v)
 {
-    std::vector<Expression*>::const_iterator i = args.begin();
+    auto i = args.begin();
```
Comment (chennes) [inline src/App/Expression.cpp:2465]: Lost `const`-ness

**Review by chennes — CHANGES_REQUESTED** [review]
I think most of this is fine, just a few minor issues (well, one major, since it will prevent compilation on Windows).

File: src/App/Expression.cpp (hunk 4 above)
Comment (berniev) [reply src/App/Expression.cpp:2465]: But looking at the entire function:
```
void FunctionExpression::_visit(ExpressionVisitor &v)
{
    auto i = args.begin();

    HiddenReference ref(f == HIDDENREF || f == HREF);
    while (i != args.end()) {
        (*i)->visit(v);
        ++i;
    }
}
```
So i can't be const. No?

File: src/App/Expression.cpp (hunk 2 above)
Comment (berniev) [reply src/App/Expression.cpp:1586]: No , I didn't. Thank you.

File: src/App/Enumeration.cpp (hunk 1 above)
Comment (berniev) [reply src/App/Enumeration.cpp:154]: Grrr. Thanks. IDE is hiding stuff.

**Comment by berniev** [comment]
Thanks! Added commit.

File: src/App/Expression.cpp (hunk 4 above)
Comment (chennes) [reply src/App/Expression.cpp:2465]: `i` can't be `const`, but its type can be a `const_iterator`. Try using `std::as_const`, something like (untested):
```
auto i = std::as_const(args).begin();
```

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6852">pipeline status</a> for feature branch [PR_6852](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6852). Pipeline [534077124](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/534077124) was triggered at [8886cc6](https://github.com/FreeCAD/FreeCAD/pull/6852/commits/8886cc6c30b5e7290fdce85e6afcfd8ce5e1b179). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/App/Expression.cpp (hunk 4 above)
Comment (berniev) [reply src/App/Expression.cpp:2465]: What if use `args.cbegin()` to make a const_iterator?

Comment (chennes) [reply src/App/Expression.cpp:2465]: Yep.

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6852">pipeline status</a> for feature branch [PR_6852](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6852). Pipeline [538023309](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/538023309) was triggered at [05dad3a](https://github.com/FreeCAD/FreeCAD/pull/6852/commits/05dad3a14ad985f44a45a7fcd7c9075fb14e57ea). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Pull request closed without merging**